armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -1233,8 +1232,16 @@ def createRepresentativeBlocksUsingExistingBlocks(
         for newXSID in modifiedReprBlocks:
             oldXSID = origXSIDsFromNew[newXSID]
             oldBlockCollection = blockCollectionByXsGroup[oldXSID]
+            if len(oldBlockCollection._validRepresentativeBlockTypes) > 0:
+                validBlockTypes = []
+                for flag in oldBlockCollection._validRepresentativeBlockTypes:
+                    validBlockTypes.append(flags._toString(Flags, flag))
+            else:
+                validBlockTypes = None
             newBlockCollection = oldBlockCollection.__class__(
-                oldBlockCollection.allNuclidesInProblem
+                oldBlockCollection.allNuclidesInProblem,
+                validBlockTypes=validBlockTypes,
+                averageByComponent=oldBlockCollection.averageByComponent,
```

Comment:
This is the business logic you are trying to implement with this ticket. And, on first glance, I think it looks good.

But I had to read it a couple times to figure out what you are doing. And, of course, I am looking at a PR with a good description.

What that tells me is this complicated logic you added probably needs at least a one-line comment telling people what they are about to read.

Does that make sense? It's an opinion, but I think the logic is complicated enough it should have a comment. (Alternatively, I suppose, some people would say it could be moved to another function with a helpful name. Meh.)

Reply:
Good point, I added a comment to describe what's happening in the new block of code